### PR TITLE
Fix unnecessary errors when redirecting on Setup page

### LINKED ISSUE #1422: ASF-ui redirects to Setup page with errors
## Description
When restarting ASF via restart button, ASF-ui redirects to Setup page with lots of errors, e.g. flags panel does not load.
https://i.imgur.com/AII9mIj.gif

## Steps to reproduce
In Chrome:
1. Clear local and session storages, remove cookies,
2. Empty cache and hard reload UI's page,
3. Open console,
4. Press restart button and open flags panel while still on Setup page.

## Screenshots
<details>
  <summary>Screenshot with errors</summary>
  
image

</details>

### PR DESCRIPTION
**Pull request by Aareksio**
## Description

- handle navigation abort by home alias redirect
- handle endless loop caused by defaultView set to "home"
- handle repetitive navigation to Setup page while already on Setup page

## Additional information

Closes #1422

## Checklist

- [x] My pull request is not a duplicate
- [x] I added a descriptive title to this pull request
- [x] I added a concise description or a self-explanatory screenshot to this pull request
- [x] My code follows the code style of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by Botan626** [comment]
UI is stuck after restart.

image

image

**Comment by Botan626** [comment]
After CTRL+F5:
image

**Comment by Botan626** [comment]
Also
image

File: src/router/routes.js [hunk 1]
```diff
@@ -14,28 +14,33 @@ export default [
...
         if (defaultView === '_last-visited-page') defaultView = storage.get('last-visited-page', { name: 'bots' });
         const page = (typeof defaultView === 'string') ? { name: defaultView } : defaultView;
-        next(page);
-			}
-			
-			next({ name: 'bots' });
+        return next(page);
+      }
+
+      next({ name: 'bots' });
```
Comment (MrBurrBurr) [inline src/router/routes.js:35]: Why not also use return here?

Comment (Aareksio) [reply src/router/routes.js:35]: The return in L32 is meant to exit function short. Here it is the last statement, hence return is not required - we do not rely on return value anyway. You can add return there :)

**Pull request merged**